src/gui/iconutils.cpp

```diff
@@ -1,52 +1,140 @@
+/*
+ * Copyright (C) by Oleksandr Zolotov <alex@nextcloud.com>
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY
+ * or FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License
+ * for more details.
+ */
+
 #include "iconutils.h"
 
 #include <theme.h>
 
 #include <QFile>
+#include <QLoggingCategory>
 #include <QPainter>
 #include <QPixmapCache>
 #include <QSvgRenderer>
 
+namespace {
+QString findSvgFilePath(const QString &fileName, const QStringList &possibleColors)
+{
+    const QString baseSvgNoColor{QString{OCC::Theme::themePrefix} + fileName};
+    if (QFile::exists(baseSvgNoColor)) {
+        return baseSvgNoColor;
+    }
+
+    for (const auto &color : possibleColors) {
+        const QString baseSVG{QString{OCC::Theme::themePrefix} + color + QLatin1Char('/') + fileName};
+
+        if (QFile::exists(baseSVG)) {
+            return baseSVG;
+        }
+    }
+
+    return {};
+}
+}
+
 namespace OCC {
 namespace Ui {
-namespace IconUtils {
-QPixmap pixmapForBackground(const QString &fileName, const QColor &backgroundColor)
-{
-    Q_ASSERT(!fileName.isEmpty());
+    Q_LOGGING_CATEGORY(lcIconUtils, "nextcloud.gui.iconutils", QtInfoMsg)
+    namespace IconUtils {
+        QPixmap pixmapForBackground(const QString &fileName, const QColor &backgroundColor)
+        {
+            Q_ASSERT(!fileName.isEmpty());
+
+            const auto pixmapColor = backgroundColor.isValid()
+                && !Theme::isDarkColor(backgroundColor)
+                ? QColorConstants::Svg::black
+                : QColorConstants::Svg::white;
+
+            return createSvgPixmapWithCustomColor(fileName, pixmapColor);
+        }
+
+        QPixmap createSvgPixmapWithCustomColor(const QString &fileName, const QColor &customColor, const QSize &size)
+        {
+            Q_ASSERT(!fileName.isEmpty());
+            Q_ASSERT(customColor.isValid());
+
+            if (fileName.isEmpty()) {
+                qWarning(lcIconUtils) << "fileName is empty";
+            }
+
+            if (!customColor.isValid()) {
+                qWarning(lcIconUtils) << "customColor is invalid";
+            }
+
+            const auto customColorName = customColor.name();
+
+            const QString cacheKey = fileName + QLatin1Char(',') + customColorName;
```

Comment:
@FlexW Not sure why? It's a char, not a string?

Reply:
@FlexW Ok. Matthieu convinced me. No more `QLatinChar('')` in this code. And, let's find some time to replace existing ones to avoid future confusion.